src/Illuminate/Database/Query/Builder.php

```diff
@@ -1388,7 +1388,14 @@ public function paginate($perPage = 15, $columns = ['*'])
 	{
 		$page = Paginator::resolveCurrentPage();
 
-		$total = $this->getCountForPagination();
+		if ($this->groups)
```

Comment:
Also repeated, should be a function. This function shall be called here and also from Builder->paginate